robot/components/trajectory_follower.py

```diff
@@ -0,0 +1,78 @@
+import pathfinder as pf
+from robotpy_ext.common_drivers import navx
+from typing import Tuple, List, Dict
+import wpilib
+from wpilib import drive
+
+
+class TrajectoryFollower:
+    """
+    Move along generated paths for autonomous
+    """
+    # TODO FIND THE REAL VALUES
+    WHEEL_DIAMETER = 0.5
+    KV = 1.101
+    KA = 0.225 # 0.102
+
+    drivetrain: drive.DifferentialDrive
+    navx: navx.AHRS
+    l_encoder: wpilib.Encoder
+    r_encoder: wpilib.Encoder
+    generated_trajectories: Dict
+
+    def on_enable(self):
+        self.last_difference = 0
+
+        self.left_follower = pf.followers.EncoderFollower(None)
+        self.right_follower = pf.followers.EncoderFollower(None)
+
+        self.left_follower.configurePIDVA(1.0, 0, 0, 1 / 10.903, 0)
+        self.right_follower.configurePIDVA(1.0, 0, 0, 1 / 10.903, 0)
+
+        self._cofigure_encoders()
+
+    def follow_trajectory(self, trajectory_name: str):
+        self._current_trajectory = trajectory_name
+        self.left_follower.setTrajectory(self.generated_trajectories[trajectory_name][0])
+        self.right_follower.setTrajectory(self.generated_trajectories[trajectory_name][1])
+
+        self._cofigure_encoders()
+
+    def _cofigure_encoders(self):
+        self.left_follower.configureEncoder(self.l_encoder.get(), 360, self.WHEEL_DIAMETER)
+        self.right_follower.configureEncoder(self.r_encoder.get(), 360, self.WHEEL_DIAMETER)
+
+    def is_following(self, trajectory_name):
+        if self._current_trajectory == trajectory_name:
```

Comment:
You should change this to `if self._current_trajectory is not None` so that passing in `None` Before you’ve given it a trajectory to follow doesn’t return True. Edge case, but good practice